test/unit/mdc-list/foundation.test.js

```diff
@@ -843,13 +851,65 @@ test('#setSelectedIndex removes selected/activated class name and sets aria-sele
   const {foundation, mockAdapter} = setupTest();
 
   td.when(mockAdapter.getListItemCount()).thenReturn(4);
+  td.when(mockAdapter.getAttributeForElementIndex(2, strings.ARIA_CURRENT)).thenReturn(null);
+  td.when(mockAdapter.getAttributeForElementIndex(3, strings.ARIA_CURRENT)).thenReturn(null);
   foundation.setSelectedIndex(2);
 
   foundation.setSelectedIndex(3);
   td.verify(mockAdapter.removeClassForElementIndex(2, cssClasses.LIST_ITEM_SELECTED_CLASS), {times: 1});
   td.verify(mockAdapter.setAttributeForElementIndex(2, strings.ARIA_SELECTED, 'false'), {times: 1});
 });
 
+test('#setSelectedIndex should detect the presence of aria-current during list initialization '
+    + '(i.e., when it is in unset state) and sets the same attribute on pre-selected index', () => {
+  const {foundation, mockAdapter} = setupTest();
+
+  td.when(mockAdapter.getListItemCount()).thenReturn(5);
+  td.when(mockAdapter.getAttributeForElementIndex(2, strings.ARIA_CURRENT)).thenReturn('page');
+  foundation.setSelectedIndex(2);
+
+  td.verify(mockAdapter.setAttributeForElementIndex(2, strings.ARIA_CURRENT, 'false'), {times: 0});
+  td.verify(mockAdapter.setAttributeForElementIndex(2, strings.ARIA_CURRENT, 'page'), {times: 1});
+});
+
+test('#setSelectedIndex should set aria-selected as default option in the absence of aria-selected on pre-selected '
```

Comment:
I'm confused looking at these two tests because both seem to have identical setup, different verifications, but both...pass? If aria-current is found, shouldn't aria-selected _not_ be set?

Reply:
The next test was fixed in immediate next commit:
https://github.com/material-components/material-components-web/pull/4479/commits/9b7f617ae2cc7ded0f0e704a86aaccf4696eabf2

You'll not see the same test setup in full PR diff. Sorry for the confusion.

Reply:
Nevermind, I overlooked the `'page'` / `null` difference.